tests/test_api.rs

```diff
@@ -9702,6 +9702,60 @@ try {
   assert!(scope.exception().is_none());
 }
 
+// Regression test for https://github.com/denoland/deno/issues/19021
+#[test]
+fn bubbling_up_exception_in_function_call() {
+  let _setup_guard = setup::parallel_test();
+  let isolate = &mut v8::Isolate::new(Default::default());
+  let scope = &mut v8::HandleScope::new(isolate);
+  let context = v8::Context::new(scope);
+  let scope = &mut v8::ContextScope::new(scope, context);
+
+  fn boom_fn(
+    scope: &mut v8::HandleScope,
+    _args: v8::FunctionCallbackArguments,
+    _retval: v8::ReturnValue,
+  ) {
+    let msg = v8::String::new(scope, "boom").unwrap();
+    let exception = v8::Exception::type_error(scope, msg);
+    scope.throw_exception(exception);
+  }
+
+  let global_proxy = scope.get_current_context().global(scope);
+  let identifier = v8::String::new(scope, "boom").unwrap();
+  let value = v8::FunctionTemplate::new(scope, boom_fn)
+    .get_function(scope)
+    .unwrap();
+  global_proxy.set(scope, identifier.into(), value.into());
+
+  let code = r#"
+(function () {
+  return function callBoom() {
+    try {
+        boom()
+    } catch (e) {
+        //
+    }
+  }
+})();
```

Comment:
I think you can just do
```js
 function callBoom() {
    try {
        boom()
    } catch (e) {
        //
    }
  }
 ```

Reply:
No, it doesn't work

Reply:
However, this *would* work:

```js
(function callBoom() {
    try {
        boom()
    } catch (e) {
        //
    }
  })
  ```
   
That's because, to avoid ambiguity between function expressions and function declarations, [an expression cannot start with the keyword `function`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/function#sect1). However it can start with a `(`, obviously!